src/tests/rollback-inconsistency.cc

```diff
@@ -0,0 +1,161 @@
+/* -*- mode: C++; c-basic-offset: 4; indent-tabs-mode: nil -*- */
+// vim: ft=cpp:expandtab:ts=8:sw=4:softtabstop=4:
+#ident "$Id$"
+/*======
+This file is part of PerconaFT.
+
+
+Copyright (c) 2006, 2015, Percona and/or its affiliates. All rights reserved.
+
+    PerconaFT is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License, version 2,
+    as published by the Free Software Foundation.
+
+    PerconaFT is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with PerconaFT.  If not, see <http://www.gnu.org/licenses/>.
+
+----------------------------------------
+
+    PerconaFT is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Affero General Public License, version 3,
+    as published by the Free Software Foundation.
+
+    PerconaFT is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Affero General Public License for more details.
+
+    You should have received a copy of the GNU Affero General Public License
+    along with PerconaFT.  If not, see <http://www.gnu.org/licenses/>.
+======= */
+
+#ident "Copyright (c) 2006, 2015, Percona and/or its affiliates. All rights reserved."
+
+#include "test.h"
+
+// insert enough rows with a child txn and then force an eviction to verify the rollback
+// log node is in valid state.
+// the test fails without the fix (and of course passes with it).
+// The test basically simulates the test script of George's. 
+
+
+static void
+populate_table(int start, int end, DB_TXN * parent, DB_ENV * env, DB * db) {
+    DB_TXN *txn = NULL;
+    int r = env->txn_begin(env, parent, &txn, 0); assert_zero(r);
+    for (int i = start; i < end; i++) {
+        int k = htonl(i);
+        char kk[4]; 
+        char str[220];
+        memset(kk, 0, sizeof kk);
+        memcpy(kk, &k, sizeof k);
+        memset(str,'a', sizeof str-1);
```

Comment:
Yes, some of those tests (especially the old ones) were messy...

-Bradley

On Fri, Oct 9, 2015 at 1:48 PM, Jun Yuan <notifications@github.com> wrote:

> In src/tests/rollback-inconsistency.cc
> <https://github.com/percona/PerconaFT/pull/332#discussion_r41658095>:
>
> > +// log node is in valid state.
> > +// the test fails without the fix (and of course passes with it).
> > +// The test basically simulates the test script of George's.
> > +
> > +
> > +static void
> > +populate_table(int start, int end, DB_TXN * parent, DB_ENV * env, DB * db) {
> > +    DB_TXN *txn = NULL;
> > +    int r = env->txn_begin(env, parent, &txn, 0); assert_zero(r);
> > +    for (int i = start; i < end; i++) {
> > +        int k = htonl(i);
> > +        char kk[4];
> > +        char str[220];
> > +        memset(kk, 0, sizeof kk);
> > +        memcpy(kk, &k, sizeof k);
> > +        memset(str,'a', sizeof str-1);
>
> Hi, Thanks @BohuTANG <https://github.com/BohuTANG> and @kuszmaul
> <https://github.com/kuszmaul> for pointing out problems. I actually
> copied the part of key and value from other code ... I should have been
> more careful with it. Will correct these issue together with my next pull
> request for fixing read-commit bug. Thanks.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/percona/PerconaFT/pull/332/files#r41658095>.
>